frontend/src/views/UsersList.vue

```diff
@@ -0,0 +1,193 @@
+        <td class="text-xs-right"> {{ props.item.location }}</td>
+        <td class="text-xs-right"> {{ props.item.emailAddress }}</td>
+        <td class="text-xs-right"> {{ props.item.phoneNumber }}</td>
+        <td class="text-xs-right"> {{ props.item.languages }}</td>
+
+        <td class="text-xs-right">
+          <v-edit-dialog
+            :return-value.sync="props.item.roles"
+            lazy
+            @save="save"
+            @cancel="cancel"
+            @open="open"
+            @close="close"
+          > {{ props.item.roles }}
+            <v-switch v-for="(role, index) in roles" :key="index"
+              slot="input"
+              :label="role"
+              :value="role"></v-switch>
+          </v-edit-dialog>
+        </td>
+        <td class="text-xs-right"> {{ props.item.experience }}</td>
+        <td class="text-xs-right"> {{ props.item.influencer }}</td>
+        <td class="text-xs-right"> {{ props.item.recommendedBy }}</td>
+      </template>
+      <template slot="table-no-data">
+        <div>
+          <div :style="{ justifyContent: 'center' }" class="d-flex" v-if="isLoadingLists">
+            <v-progress-circular indeterminate></v-progress-circular>
+          </div>
+        </div>
+        </template>
+    </v-data-table>
+
+    <v-snackbar v-model="snack" :timeout="5000" :color="snackColor">
+      {{ snackText }}
+      <v-btn flat @click="snack = false">Close</v-btn>
+    </v-snackbar>
+  </div>
+</template>
+
+<script>
+import DataTable from "@/components/DataTable.vue";
+
+export default {
+  name: "UsersList",
+  data() {
+    return {
+      users: [],
+      isLoadingLists: false,
+      snack: false,
+      snackColor: "",
+      snackText: "",
+      pagination: {},
+      statusItems: ["Empty", "Opted out", "Lost", "Duplicate", "Incorrect"],
+      status: ["Empty", "Opted out", "Lost", "Duplicate", "Incorrect"],
+      roles: ["CR", "TE", "SE", "QC", "FC", "SQR Coordinator"],
+      max25chars: v => v.length <= 25 || "Input too long!",
+      headers: [
+        { text: "Notes", value: "notes" },
+        { text: "Status", value: "status" },
+        { text: "Timestamp", value: "timestamp" },
+        { text: "Name", value: "name" },
+        { text: "Location", value: "location" },
+        { text: "Email Address", value: "emailAddress" },
+        { text: "Phone Number", value: "phoneNumber" },
+        { text: "Languages", value: "languages" },
+        { text: "Roles", value: "roles" },
+        { text: "Experience", value: "experience" },
+        { text: "Influencer", value: "influencer" },
+        { text: "Recommended By", value: "recommendedBy" }
+      ]
+    };
+  },
+  components: {
+    DataTable
+  },
+  methods: {
+    async retrieveUsers() {
+      let userObj = null;
+      let usersList = [];
+      let lang = "";
+      let langString = "";
+      let roles = "";
+      let rolesString = "";
+      this.isLoadingLists = true;
+      const response = await this.$http.get(
```

Comment:
[Vuefire](https://github.com/vuejs/vuefire/tree/v1) is currently in this project and should be used.